dotnet/src/SemanticKernel.Core/KernelExtensions.cs

```diff
@@ -701,7 +703,34 @@ public static IKernelBuilderPlugins AddFromFunctions(this IKernelBuilderPlugins
             promptTemplateFactory: promptTemplateFactory,
             loggerFactory: kernel.LoggerFactory);
 
-        return kernel.InvokeAsync<T>(function, arguments);
+        return kernel.InvokeAsync<T>(function, arguments, cancellationToken);
+    }
+
+    /// <summary>
+    /// Invokes a prompt specified via a prompt template and returns the results of type <typeparamref name="T"/>.
+    /// </summary>
+    /// <param name="kernel">The <see cref="Kernel"/> containing services, plugins, and other state for use throughout the operation.</param>
+    /// <param name="promptTemplate">Prompt template for the function.</param>
+    /// <param name="arguments">The arguments to pass to the function's invocation, including any <see cref="PromptExecutionSettings"/>.</param>
+    /// <param name="templateFormat">The template format of <paramref name="promptTemplate"/>. This must be provided if <paramref name="promptTemplateFactory"/> is not null.</param>
+    /// <param name="promptTemplateFactory">
+    /// The <see cref="IPromptTemplateFactory"/> to use when interpreting the <paramref name="promptTemplate"/> into a <see cref="IPromptTemplate"/>.
+    /// If null, a default factory will be used.
+    /// </param>
+    /// <returns>The <typeparamref name="T"/> of the function result value.</returns>
+    /// <exception cref="ArgumentNullException"><paramref name="kernel"/> is null.</exception>
+    /// <exception cref="ArgumentNullException"><paramref name="promptTemplate"/> is null.</exception>
+    /// <exception cref="ArgumentException"><paramref name="promptTemplate"/> is empty or composed entirely of whitespace.</exception>
+    /// <exception cref="KernelFunction">The function failed to invoke successfully.</exception>
+    /// <exception cref="KernelFunctionCanceledException">The <see cref="KernelFunction"/>'s invocation was canceled.</exception>
+    public static Task<T?> InvokePromptAsync<T>(
```

Comment:
I would propose to mark this method as obsolete since it does not provide any benefits and was introduced only to avoid the breaking change. We will remove all obsolete stuff in SK V2. @stephentoub , @markwallace-microsoft thoughts?

Reply:
> We will remove all obsolete stuff in SK V2.

No, not unless you want to break every binary referencing that thing.

Reply:
The compiler will bind to this overload of a developer passes in an argument for every parameter. Marking it obsolete would give such call sites obsoletion warnings. We can hide it from intellisense with EditorBrowsable, though.

Reply:
(Welcome to the wonderful world of compatibility.)

Reply:
@stephentoub! So, if I’m getting this right, we could consider removing it for SK v2. But, marking it as obsolete right now might not be the best move as it could trigger some unnecessary warnings. Handling it with `EditorBrowsable` seems to be a more suitable method to manage this situation. Did I get that right? If so, I will apply this attribute.

Reply:
We can't delete it and we can't mark it obsolete. Best we can do is hide it with editor browsable.

Reply:
@stephentoub I’ve just added the `EditorBrowsable` attribute and that neat little overload mentioned in this [comment](https://github.com/microsoft/semantic-kernel/issues/4573#issuecomment-1889668110) by @bartonjs. This should make everything line up nicely with the Framework Design Guidelines (FDG).

Reply:
We have a ton of methods in SK that don't follow that guideline. I think that's ok, and we shouldn't be special-casing this one with such an additional overload.

Reply:
Agree, just removed the extra one.